arbor/fvm_lowered_cell_impl.hpp

```diff
@@ -913,17 +915,27 @@ void resolve_probe(const cable_probe_point_state& p, probe_resolution_data<B>& R
     // Convert cell-local target number to cellgroup target number.
     const auto& divs = R.M.target_divs;
     auto cell = R.cell_idx;
-    auto cg  = target + divs.at(cell);
-    if (cg >= divs.at(cell + 1)) return;
-
-    const auto& handle = R.handles.at(cg);
-    if (handle.mech_id != mech_id) return;
-    auto mech_index = handle.mech_index;
-    R.result.push_back(fvm_probe_scalar{{data + mech_index},
-                       point_info_of(target,
-                                     mech_index,
-                                     synapses.at(mech),
-                                     R.M.mechanisms.at(mech).multiplicity)});
+    auto cg_lo = divs.at(cell);
+    auto cg_hi = divs.at(cell + 1);
+    const auto& [lr_beg, lr_end] = R.cell
+                                    .synapse_ranges()
+                                    .equal_range(t_hash);
+    for (auto lr = lr_beg; lr != lr_end; ++lr) {
+        const auto& [lid_beg, lid_end] = lr->second;
+        for (auto lid = lid_beg; lid != lid_end; ++lid) {
+            auto cg = lid + cg_lo;
+            if (cg >= cg_hi) continue;
+            const auto& handle = R.handles.at(cg);
+            if (handle.mech_id != mech_id) return;
+            auto mech_index = handle.mech_index;
+            R.result.push_back(fvm_probe_scalar{{data + mech_index},
+                                                 point_info_of(target,
+                                                               lid,
+                                                               mech_index,
+                                                               synapses.at(mech),
+                                                               R.M.mechanisms.at(mech).multiplicity)});
+        }
+    }
```

Comment:
just for my understanding - can there ever be more than one probe resolved and pushed to result?

Reply:
Formally, yes, as `synapse_ranges` is a multimap of ranges. This _could_ happen 
if two synapses are given the same label. We could forbid that, but currently don't.

Reply:
sure, sounds good

Reply:
Ah, that's impossible, as the probe contains the mechanism type (the id in the resolution step):
```
if (handle.mech_id != mech_id) return; 
```
So, multiple synapses of the **same** type can appear in the same probe, here.